### DISCUSSION TITLE
tsml_scan_data_source Cron Job

### DISCUSSION THREAD

**Discussion by Nickjones818** [code4recovery/12-step-meeting-list#748] [Q&A] [answered]
I'm running into an issue where tsml_scan_data_source() is a part of our scheduled actions in WP_control, and it seems to be running daily per the date stamps, but it's not actually refreshing the Data source in TSML. Any ideas on what might be happing?

**Comment by joshreisner** [comment, marked as answer]
hi @Nickjones818 - i believe that job only scans the data source and notifies you by email if there's a change, it doesn't actually refresh the data source automatically

**Reply by Nickjones818** [reply]
I don't seem to be receiving emails it looks like, I do have SMTP mail setup on this Site, but have checked all possible Mailboxes just in case and it doesn't seem to be going through. I'd rather it just be automated as opposed to notified anyway, lol. Is it possible to run Cron Job that would actually refresh the Feed?

**Reply by joshreisner** [reply]
i know right? problem is importing requires geocoding, which relies on an external service, so it's too complex for wordpress to handle in the background.

we should try to get the email thing working at least! you already answered my first question, which is going to be is SMTP working. can you confirm that the website can send emails? for example password reset emails?

next, let's confirm that the data source says "Enabled" in the change detection column.

third: do you have email addresses specified under "Change Notification Emails" on the right side?

finally, confirm a change to the data sources

then, theoretically, if all are true, you should receive an email within 24 hours of the data source changing

**Reply by Nickjones818** [reply]
That's a bummer on the Geocoding roadblock, maybe another option will appear someday.

I believe #3 was my problem 🤦‍♂️. I am getting emails from the site on a regular basis through our contact forms and through our Wordfence installation. Also, change detection is enabled on our data feed. Our Central office will be making another update to their list this evening so ill verify a lack of an email address to send notifications to was the problem.  I appreciate the help, Josh.
